core/src/main/scala/org/apache/spark/sql/rapids/tool/util/ToolsPlanGraph.scala

```diff
@@ -272,13 +273,16 @@ object ToolsPlanGraph {
   }
 
   private def buildSparkPlanGraphNode(
-      planInfo: SparkPlanInfo,
+      planInfoRaw: SparkPlanInfo,
       nodeIdGenerator: AtomicLong,
       nodes: mutable.ArrayBuffer[SparkPlanGraphNode],
       edges: mutable.ArrayBuffer[SparkPlanGraphEdge],
       parent: SparkPlanGraphNode,
       subgraph: SparkPlanGraphCluster,
       exchanges: mutable.HashMap[SparkPlanInfo, SparkPlanGraphNode]): Unit = {
+    // Replace Photon node names with Spark node names
+    // TODO: Skip this if app.isPhoton is false
```

Comment:
I would like to see this sooner rather then later if we can easily determine up front

Reply:
Changed the flow. We do not create a new instance of `SparkPlanInfo` here. We are only interested in the `name`